modules/core/src/v2/coins/sol.ts

```diff
@@ -100,8 +99,35 @@ export class Sol extends BaseCoin {
   }
 
   // TODO (https://bitgoinc.atlassian.net/browse/STLX-10202)
-  async verifyTransaction(params: VerifyTransactionOptions): Promise<boolean> {
-    throw new MethodNotImplementedError('verifyTransaction method not implemented');
+  async verifyTransaction(params: VerifyTransactionOptions): Promise<any> {
+    let totalAmount = 0;
+    const coinConfig = coins.get(this.getChain());
+    const { txParams: txParams, txPrebuild: txPrebuild } = params;
+    const transaction = new accountLib.Sol.Transaction(coinConfig);
+    if (!txPrebuild.txBase64) {
+      throw new Error('missing required tx prebuild property txBase64');
+    }
+    transaction.fromRawTransaction(txPrebuild.txBase64);
+    const explain = transaction.explainTransaction();
+    explain.outputs.should.deepEqual(txParams.recipients);
+    const transactionJson = transaction.toJson();
+    if (txParams.memo != explain.memo) {
+      throw new Error('Diferent memo');
+    }
+    if (explain.outputs.length != txParams.recipients?.length) {
+      throw new Error('Diferent number of recipients');
+    }
+    for (let i = 0; i < txParams.recipients.length; i++) {
+      totalAmount += new BigNumber(txParams.recipients[i].amount).toNumber();
+      // totalAmount = totalAmount + txParams.recipients[i].amount;
+    }
+    if (totalAmount != new BigNumber(explain.outputAmount).toNumber()) {
+      throw new Error('Diferent total amount');
+    }
+    if (transactionJson.feePayer != txParams.feePayer) {
+      throw new Error('Diferent fee P amount');
+    }
+    return true;
```

Comment:
You need to find a way to verify each amount of each instruction inside the tx.

Reply:
with this i think i covered this 
![image](https://user-images.githubusercontent.com/85512331/147595622-7c73f2b4-c1fb-45a3-a671-e577bde9ed54.png)

Reply:
Yes agree but if the params are diff with the explainTx you need to throw an exception. So may be will be good a validation for this.